docs/timeout-activities-and-workflows.md

```diff
@@ -0,0 +1,119 @@
+| `StartToClose`    | Duration | time.Time | Maximum time an Activity can take to  execute after it was picked by a Worker. | TODO |
+| `ScheduleToClose` | Duration | time.Time | Maximum time from when the Workflow requests an Activity execution to its completion including retries | TODO |
+
+### Example usage
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+    defaultValue="java"
+    values={[
+        { label: 'Java', value: 'java', },
+        { label: 'Go', value: 'go', },
+    ]
+}>
+
+<TabItem value="java">
+
+```java
+private final RemoteDataActivities activities = Workflow.newActivityStub(
+    RemoteDataActivity.class,
+    ActivityOptions.newBuilder()
+    .setScheduleToStartTimeout(Duration.ofMinutes(1))
+    .setScheduleToCloseTimeout(Duration.ofMinutes(20))
+    .setStartToCloseTimeout(Duration.ofMinutes(10))
+    .build()
+);
+```
+
+</TabItem>
+<TabItem value="go">
+
+```go
+sample := "TODO"
+//Go Activity timeout sample
+```
+
+</TabItem>
+</Tabs>
+
+In the above code snippets:
+
+1. We set `ScheduleToStart` to a duration of 1 minute. This means that our Workflow will wait a maximum of 1 minute for a Worker to pick up this Activity. If it is exceeded the Activity will either be retried or an error will be surfaced to the Worker. In the majority of cases it makes sense to set this to the expected run duration of the Activity.
+2. We set `ScheduleToClose` to a duration of 1 minute. This means that our Workflow will wait a maximum of 1 minute for an Activity execution to complete starting from when the Task was requested to completion, including retries. If it is exceeded an error will be surfaced to the Worker.
+3. We set `StartToCloseTimeout` to a duration of 10 minutes. This means that our Workflow will wait a maximum of 10 minutes for a Worker that has picked up the Activity to finish executing it. If it is exceeded the Activity will either be retried or an error will be surfaced to the Worker.
+
+## Workflow timeouts
+
+Activity timeouts are great when you need to impose time restrictions on a specific Activity, but there's often a need to impose similar restrictions on a workflow execution. The code snippet below shows all the timeout properties that are available to workflows.
+
+### Timeout policy options
+
+The following field values can be customized within the `WorkflowOptions`:
+
+| Field | Java Type | Go Type | Description | Default |
+|-------|-----------|---------|-------------|---------|
+| `WorkflowExecutionTimeout` | Duration | time.Time | Maximum time for the Workflow Execution including retries and "ContinueAsNew"". Use WorkflowRunTimeout to limit execution time of a single workflow run. | TODO |
+| `WorkflowTaskTimeout`      | Duration | time.Time | Maximum time from when a Worker pulls the task. If a Workflow Task is lost, it is retried after this timeout. | TODO |
```

Comment:
WorkflowTaskTimeout defines how long the workflow is going to get "stuck" in the case of a worker crash. So setting it to a large value is not a good idea. Setting it too low (below 10 seconds) is not usually recommended as it can lead to unnecessary timeouts which might increase the load on the service and worker processes.